### ISSUE TITLE
Enable full domain configuration also for Default Policy: [*] Allow requests to the same domain

### ISSUE THREAD

**Issue by 360up** [RequestPolicyContinued/requestpolicy#688]
> about:requestpolicy?defaultpolicy
> - [x] Allow requests to the same domain (www.example.com -> static.example.com).

When the above default policy is set the tool bar only allows to configure requests to a whole domain tree (eg *.example.com) but not to an exact full domain (eg: cdn1.example.com). Further it only allows 3rd-party configuration.

It's only a default policy, so it should not remove the possibility to set custom full domain rules altogether. I am aware you can still manually add those in the settings, but it'll be nice to do so via tool bar too. Not only to allow full 3rd-party domains, but also to block some unwanted 1st-party.

**Comment by nodiscc** [comment]
Hi @360up Is this the issue you're looking for? https://github.com/RequestPolicyContinued/requestpolicy/issues/474 or https://github.com/RequestPolicyContinued/requestpolicy/issues/454

**Comment by 360up** [comment]
> Is this the issue you're looking for? #474 or #454

No, this here is about the limited control in the toolbar when the above mentioned default policy is set.

For example options for requests to ytimg.com on youtube.com are limited to
` *.ytimg.com `

instead of additional full domain options
` s.ytimg.com `
` i.ytimg.com `
` *.ytimg.com `

**Comment by myrdd** [comment]
Indeed. This issue is related to the *„strictness“*. As of RP v0.5 a *"base domain"* strictness always displayed the "base domain" as well. It was not possible to display the "full domain" with the "base domain" strictness. — By the way, even the rules have been strictness-specific, i.e. "full domain" rules only applied in the "full domain" strictness mode.

--

@360up do I understand correctly – you'd like to *uncouple* the strictness of implying what is displayed?

So there would be two different options: "strictness" and "displayed strictness" (or "strictness in the menu"). Both options would have the selections "base domain" and "full domain".

Those are the possible combinations:

| strictness | displayed strictness |
| ------------ | ------------------------- |
| base domain | base domain |
| base domain | full domain |
| full domain | full domain |

The fourth combination `full domain / base domain` wouldn't make sense.

---

Thanks for that hint @360up. I think it's now more clear what needs to be done with the menu.

**Comment by 360up** [comment]
> do I understand correctly – you'd like to uncouple the strictness of implying what is displayed?

Yes, full domain configuration also for base domain strictness (both 1st and 3rd party). At least as an optional feature.



> Thanks for that hint @360up. I think it's now more clear what needs to be done with the menu.

So can we expect that being implemented in a future release?

**Comment by myrdd** [comment]
Okay @360up. As it's now clear to me what needs to be done, I've changed the title of this issue. When I've before been talking about „displayed strictness“, I think „(domain-level) configurability“ is the better wording.

> So can we expect that being implemented in a future release?

Yes. I'm not sure if it will make it into 1.0 though. This issue depends on the implementation of the strictness, #474.

**Comment by myrdd** [comment]
Maybe the configuration should be even more detailed than only „strictness“ and „configuration“.

* **default policy**
  * allow
  * or deny – then choose the **strictness** – „How much is allowed by default in default-deny mode?“
    * the same base domain
    * the same host (full domain)
    * or same pre-path (full address, that is, domain + protocol + port)
* **menu configurability** – left and right column
  * **left column** – „What's the level of detail displayed?“
    * base domain
    * full domain
    * full address
  * **right column** – „Which types of rules can be added?“ – multiple choices possible
    * Case A: both *origin* and *destination* are selected on the left column
      * `* --> base domain`
      * `base domain --> base domain`
      * `* --> full domain`
      * `full domain --> full domain`
      * `* --> pre-path`
      * `pre-path --> pre-path`
    * Case B: only an *origin* is selected on the left column
      * `base domain --> *`
      * `full domain --> *`
      * `pre-path --> *`

Does this make sense?

---

So the full settings page – leaving out descriptions – could look like this:

* **default policy**
  * `allow`
  * or `deny`
    * **strictness** – single choice
      * `base domain`
      * `full domain`
      * `full address`
* [ ] **configure the menu** – optional
  * **left column** – single choice
    * „show *base domain*“
    * „show *full domain*“
    * „show *full address*“
  * **right column** – multiple choice
    * [ ] „show *base domain* rules“
    * [ ] „show *full domain* rules“
    * [ ] „show *full address* rules“

---

As the default for new installations I suggest:

* **default policy:** deny – see also #614
* **strictness:** base domain
* **left menu column:** base domain
* **right menu column:** show base domain only

By default we'll have: `strictness === left col === right col`.

What do you think?
